lib/src/components/navigation bar/navigation_bar.dart

```diff
@@ -0,0 +1,277 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// An item to be used in a [ZetaNavigationBar]
+class ZetaNavigationBarItem {
+  /// Creates a new [ZetaNavigationBarItem]
+  const ZetaNavigationBarItem({
+    required this.icon,
+    required this.label,
+    this.showBadge = false,
+    this.badgeValue,
+  });
+
+  /// The icon shown on the item.
+  final IconData icon;
+
+  /// The label shown on the item.
+  final String? label;
+
+  /// Displays a small [ZetaIndicator] on the item.
+  final bool showBadge;
```

Comment:
To match other components, I think we should change this flag to being the actual Badge widget, and hardcode it to being the correct size.